pkg/controller/backupbucket/actuator_test.go

```diff
@@ -0,0 +1,170 @@
+	. "sigs.k8s.io/controller-runtime/pkg/envtest/komega"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	controllerconfig "github.com/onmetal/gardener-extension-provider-onmetal/pkg/apis/config"
+	"github.com/onmetal/gardener-extension-provider-onmetal/pkg/onmetal"
+	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Backupbucket Reconcile", Ordered, func() {
+	var (
+		backupBucket *extensionsv1alpha1.BackupBucket
+		bucket       *storagev1alpha1.Bucket
+	)
+	ns := SetupTest()
+
+	It("should create backup bucket for a given bucket configuration", func(ctx SpecContext) {
+		By("creating backup bucket resource")
+		backupBucket = &extensionsv1alpha1.BackupBucket{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns.Name,
+				Name:      "my-backup-bucket",
+			},
+			Spec: extensionsv1alpha1.BackupBucketSpec{
+				DefaultSpec: extensionsv1alpha1.DefaultSpec{
+					Type:           onmetal.Type,
+					ProviderConfig: nil,
+				},
+				Region: "europe-central",
+				SecretRef: corev1.SecretReference{
+					Name:      "backupprovider",
+					Namespace: ns.Name,
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, backupBucket)).Should(Succeed())
+
+		Eventually(Object(backupBucket)).Should(SatisfyAll(
+			HaveField("Status.LastOperation.Type", gardencorev1beta1.LastOperationTypeCreate),
+		))
+
+		By("ensuring backup bucket is created with correct spec")
+
+		bucket = &storagev1alpha1.Bucket{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns.Name,
+				Name:      "my-backup-bucket",
+			},
+		}
+
+		Eventually(Object(bucket)).Should(SatisfyAll(
+			HaveField("Spec.BucketClassRef", &corev1.LocalObjectReference{
+				Name: "my-bucket-class",
+			}),
+		))
+
+		bucketAccesSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns.Name,
+				Name:      "my-bucket-secret",
+			},
+			Data: map[string][]byte{
+				"AWS_ACCESS_KEY_ID":     []byte("YXdzQWNjZXNzS2V5Cg=="),
+				"AWS_SECRET_ACCESS_KEY": []byte("YXdzQWNjZXNzU2VjcmV0S2V5Cg==ss"),
+			},
+		}
+		Expect(k8sClient.Create(ctx, bucketAccesSecret)).To(Succeed())
+
+		By("patching backup bucket with available state and access details")
+		bucketBase := bucket.DeepCopy()
+		bucket.Status.State = storagev1alpha1.BucketStateAvailable
+		bucket.Status.Access = &storagev1alpha1.BucketAccess{
+			SecretRef: &corev1.LocalObjectReference{
+				Name: "my-bucket-secret",
+			},
+			Endpoint: "endpoint-efef-ihfbd-ssadd.storage",
+		}
+		Expect(k8sClient.Status().Patch(ctx, bucket, client.MergeFrom(bucketBase))).To(Succeed())
+
+		By("ensuring that bucket is created and Available")
+		Eventually(Object(bucket)).Should(SatisfyAll(
+			HaveField("Status.State", storagev1alpha1.BucketStateAvailable),
+			HaveField("Status.Access.SecretRef.Name", "my-bucket-secret"),
+			HaveField("Status.Access.Endpoint", "endpoint-efef-ihfbd-ssadd.storage"),
+		))
+
+		Eventually(Get(bucket)).Should(Succeed())
```

Comment:
We are checking the `bucket` on top already - it should exist at this point. This line can be removed.

Reply:
removed